vortex-serde/src/layouts/read/layouts/chunked.rs

```diff
@@ -75,6 +76,66 @@ impl ChunkedLayout {
         }
     }
 
+    fn child_ranges(&self) -> VortexResult<Vec<(usize, usize)>> {
+        let children = self
+            .flatbuffer()
+            .children()
+            .ok_or_else(|| vortex_err!("Missing children"))?;
+        Ok(children
+            .iter()
+            .map(|c| c.length())
+            .scan(0u64, |acc, length| {
+                let current = *acc;
+                *acc += length;
+                Some((current as usize, *acc as usize))
+            })
+            .collect::<Vec<_>>())
+    }
+
+    fn ranged_children(&self) -> VortexResult<VecDeque<RangedLayoutReader>> {
+        self.flatbuffer()
+            .children()
+            .ok_or_else(|| vortex_err!("Missing children"))?
+            .iter()
+            .enumerate()
+            // Skip over the metadata table of this layout
+            .skip(if self.has_metadata() { 1 } else { 0 })
+            .zip_eq(self.child_ranges()?)
+            .map(|((i, c), (begin, end))| {
+                let layout = self.layout_builder.read_layout(
+                    self.fb_bytes.clone(),
+                    c._tab.loc(),
+                    self.scan.clone(),
+                    self.message_cache
+                        .relative(i as u16, self.message_cache.dtype().clone()),
+                )?;
+                Ok(((begin, end), layout))
+            })
+            .collect::<VortexResult<VecDeque<_>>>()
+    }
+
+    fn metadat_only_children(&self) -> VortexResult<Vec<RangedLayoutReader>> {
+        self.flatbuffer()
+            .children()
+            .ok_or_else(|| vortex_err!("Missing children"))?
+            .iter()
+            .enumerate()
+            // Skip over the metadata table of this layout
+            .skip(if self.has_metadata() { 1 } else { 0 })
+            .zip_eq(self.child_ranges()?)
```

Comment:
it doesn't look like `child_ranges()` skips the first child if has_metadata. Should it?

Reply:
yes this is a bug